src/lib/ts/tsconfig.ts

```diff
@@ -19,7 +17,30 @@ export function readDefaultTsConfig(fileName?: string): TsConfig {
     fileName = path.resolve(__dirname, 'conf', 'tsconfig.ngc.json');
   }
 
-  return ng.readConfiguration(fileName);
+  // these options are mandatory
+  const extraOptions: ng.CompilerOptions = {
+    moduleResolution: ts.ModuleResolutionKind.NodeJs,
+    target: ts.ScriptTarget.ES2015,
+    emitDecoratorMetadata: true,
+    experimentalDecorators: true,
+
+    // sourcemaps
+    sourceMap: false,
+    inlineSources: true,
+    inlineSourceMap: true,
+
+    outDir: '',
+    declaration: true,
+
+    // ng compiler to options
+    enableResourceInlining: true,
+
+    // these are required to set the appropriate EmitFlags
+    flatModuleId: 'AUTOGENERATED',
+    flatModuleOutFile: 'AUTOGENERATED'
+  };
+
+  return ng.readConfiguration(fileName, extraOptions);
```

Comment:
Note: By providing the extraOptions when reading the configuration, ngc can appropriate set the correct emitFlags

Reply:
Does this mechanism allow overwriting `flatModuleId`, `flatModuleOutFile` in the user tsconfig?

Reply:
Not really, but it was never the case.

Reply:
Fair enough 👍